otherlibs/fiber/pool.ml

```diff
@@ -20,26 +16,58 @@ let running t k =
   | Open -> k true
   | Closed -> k false
 
-let create () = { mvar = Mvar.create (); status = Open }
+let create () = { tasks = Queue.create (); runner = None; status = Open }
 
 let task t ~f k =
   match t.status with
   | Closed ->
     Code_error.raise "pool is closed. new tasks may not be submitted" []
-  | Open -> Mvar.write t.mvar (Task f) k
-
-let stream t =
-  Stream.In.create (fun () ->
-      let+ next = Mvar.read t.mvar in
-      match next with
-      | Done -> None
-      | Task task -> Some task)
+  | Open -> (
+    match t.runner with
+    | None ->
+      Queue.push t.tasks f;
+      k ()
+    | Some r ->
+      assert (Queue.is_empty t.tasks);
+      t.runner <- None;
+      Queue.push t.tasks f;
+      resume r () k)
 
 let stop t k =
   match t.status with
   | Closed -> k ()
```

Comment:
Currently, there is some asymmetry: calling `run` multiple times is disallowed, but calling `stop` multiple times is fine. Is this desirable? I guess multiple `stop`s may be harmless but might they be an indication of a bug? Does it make sense to forbid multiple calls to `stop`?

Reply:
Our convention is to make `stop`/`close` idempotent (rpc, Stream.Out.close, etc). Personally I like this convention.

Reply:
OK, sounds good! It would be nice to write this down somewhere in this module.